File: frontend/src/utilities/context/types/GameOverview.ts
import type {Game} from "./Game.ts";
import type {PlayerScore} from "./PlayerScore.ts";
import type {GameLeaderboard} from "./GameLeaderboard.ts";

export type GameOverview = {
    game: Game,
    leaderboard: GameLeaderboard[];
}


// UUID getPlayerId();
// String getUsername();
// Long getTotalScore();

// private Game game;
// private List<PlayerScore> leaderboard;

File: src/main/java/com/ordnance/karakal/rest/replay/objects/GameOverview.java
package com.ordnance.karakal.rest.replay.objects;

import com.ordnance.karakal.rest.replay.entities.Game;
import com.ordnance.karakal.rest.replay.entities.PlayerScore;

import java.util.List;

public class GameOverview {
    private Game game;
    private List<PlayerScore> leaderboard;

    public Game getGames() {
        return game;
    }

    public List<PlayerScore> getReplay() {
        return leaderboard;
    }

    public GameOverview(Game games, List<PlayerScore> leaderboard) {
        this.game = game;
        this.leaderboard = leaderboard;
    }
}
